### PR TITLE
Contourf skeleton

### PR DESCRIPTION
**Pull request by corinnebosley**
This is bare bones, no checks.  

When it makes the axes, it turns the cube's coordinate system into a GeoAxes but does not check whether the cube has a coord system in the first place, but this shouldn't be a problem if it is a temporary measure.

### CODE REVIEW

File: lib/cube_browser/__init__.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import matplotlib.pyplot as plt
+
+
+class Contourf(object):
```
Comment (marqh) [inline lib/cube_browser/__init__.py]: docstring for the class please
What is a Contourf?

File: lib/cube_browser/__init__.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    def __init__(self, cube, coords, **kwargs):
```
Comment (marqh) [inline lib/cube_browser/__init__.py]: docstring for the init please, list the args and kwargs

provide links to iris.plot.contourf and matplotlib.pyplot.contourf for kwarg interpretation

File: lib/cube_browser/__init__.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        self.cube = cube
+        self.coords = coords # coords to plot
+        self.kwargs = kwargs
+        self.axes = self.new_axes()
+        self.qcs = None  # QuadContourSet
+        # A mapping of 1d-coord name to dimension
+        self.coord_dim = self.coord_dims()
+
+    # This constructs the original and updated plots
```
Comment (marqh) [inline lib/cube_browser/__init__.py]: make this a docstring please

File: lib/cube_browser/__init__.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    def __call__(self, **kwargs):
```
Comment (marqh) [inline lib/cube_browser/__init__.py]: i'm a little wary of this function taking kwargs i think i prefer taking one argument, which is iterable, and not using kwargs.items()

@bjlittle 's opinion is really helpful here

File: lib/cube_browser/__init__.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        for name, value in kwargs.items():
+            if name in self.coord_dim:
+                index[self.coord_dim[name]] = value
+        cube = self.cube[tuple(index)]
+        plt.sca(self.axes)
+        self.qcs = iplt.contourf(cube, coords=self.coords,
+                                 axes=self.axes, **self.kwargs)
+        return plt.gca()
+
+    def new_axes(self):
```
Comment (marqh) [inline lib/cube_browser/__init__.py]: a comment here that this is temporary (if it is) would be helpful
have we made an issue for managing axes yet?

File: lib/cube_browser/__init__.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        fig = plt.figure()
+        ax = fig.add_subplot(111, projection=cube.coord_system().
+                             as_cartopy_projection())
+        return ax
+
+    # This makes a mapping dictionary of DIMENSION coordinates
```
Comment (marqh) [inline lib/cube_browser/__init__.py]: as a docstring please

File: lib/cube_browser/__init__.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    def coord_dims(self):
+        mapping = {}
+        for dim in range(self.cube.ndim):
+            coords = self.cube.coords(dimensions=(dim,))
+            mapping.update([(c.name(), dim) for c in coords])
+        return mapping
+
+    # This makes a list of the dim coords not used on the axes,
+    # so they are available for use as sliders
```
Comment (marqh) [inline lib/cube_browser/__init__.py]: as a docstring please

**Comment by marqh** [comment]
closes #13

File: lib/cube_browser/__init__.py (hunk 4 above)
Comment (bjlittle) [inline lib/cube_browser/__init__.py]: The choice we take here might be influenced by our approach when we come to deal with anonymous dimensions that are controlled by sliders.

So if we have ...
```python
cf = Contourf(cube, coords=('longitude', 'latitude'))
```
The current approach allows a call with ...
```python
cf(time=10, model_level_number=23)
```
as opposed to something like ...
```python
cf([('time', 10), ('model_level_number', 23)])
```
or ...
```python
cf({'time': 10, 'model_level_number':23})
```
or something else ... I think we need to team-talk this one.

Comment (corinnebosley) [inline lib/cube_browser/__init__.py]: Yep, it's the anonymous dimensions which we will have to be clear for here.  They are pesky.

Comment (corinnebosley) [inline lib/cube_browser/__init__.py]: I think I like the dictionary approach best.  I think that will be clearest.

Comment (bjlittle) [inline lib/cube_browser/__init__.py]: @corinnebosley As a temporary compromise, could you create an issue and add a cross-reference comment in the code for now :pray:

File: lib/cube_browser/__init__.py (hunk 1 above)
Comment (bjlittle) [inline lib/cube_browser/__init__.py]: @marqh What style do we want to use for doc-strings?

**Comment by corinnebosley** [comment]
@marqh 
I have changed the docstring style to be the same as that for iris, and I have added some issue cross-refs for open questions.

File: lib/cube_browser/__init__.py (hunk 1 above)
Comment (marqh) [inline lib/cube_browser/__init__.py]: i'd prefer numpy style docstrings, they're what I'm used to and common across SciTools

**Comment by marqh** [comment]
i have merged #23 

I would very much like to see a simple test of this functionality added but I don't think that's essential before we merge
#28
@bjlittle what do you think might be a minimal useful test?

I suggest
`lib/cube_browser/tests/unit/test_Contourf.py`
as a new test file

lib/cube_browser/tests/unit/test.py
provides a simplistic pattern for a test to run

**Pull request merged**